packages/instrumenter/src/transformers/babel-transformer.ts

```diff
@@ -84,6 +92,40 @@ export const transformBabel: AstTransformer<ScriptFormat> = (
     root.program.body.unshift(...header);
   }
 
+  /**
+   * Takes the leading comments of a location and adds if they're a mutation disable comment then adds them
+   * to the mutantDisableMap
+   */
+  function addCommentsToMutantDisableMap(location: types.SourceLocation | null, comments: readonly types.Comment[] | null): void {
+    if (!comments || !location) {
+      return;
+    }
+
+    comments
+      .filter((comment) => comment.value.trim().startsWith('Stryker disable-next-line'))
+      .forEach((comment) => {
+        const [_, _disableType, mutations] = comment.value.trim().split(' ');
```

Comment:
We should also keep track of the `ignoreReason`. For example:

```js
// Stryker disable-next-line update The update mutator creates an infinite loop here.
```

The ignore reason we can then report is: `"Disabled by user comment. The update mutator creates an infinite loop here."`

For this to work, and keep parsing of the comment positional, we would need to force users to use `// Stryker disable-next-line all` if they want to ignore all, otherwise we cannot tell where the ignore reason starts.

What do you think about this?

Reply:
It's not my preference, having to type `all` seems coutner to how I'd expect it to work (Personally I'd expect it to work the way eslint does since we're in a JS context and that's how I kind of expect these things to work in JS). I'd **prefer** to have have us take the positional argument there and check it against the list of enabled mutations to determine if it's a specific mutant or the start of a comment. I think we'd have to do this anwyay if we want to force the positional argument in order to check if the user accidently left it off.

Reply:
Hmm good point. But still don't want to leave it up to "chance". 

Maybe we can force the mutator names to be in square brackets? Some examples:

```js
// Stryker disable-next-line [update] The update mutator creates an infinite loop here.
// Stryker disable-next-line We don't care about log statements
// Stryker disable-next-line [update] 
// Stryker disable-next-line
```

(the last 2 are without ignoreReasons)

If there are no ignore reasons given, we should still ignore them with "Disabled by user comment", or something like that.